### PR TITLE
refactor(e2e): improve stability by discovery validation in Trading tests

### PR DESCRIPTION
**Pull request by Vere-Grey**
I recently found out there was an edge case where discovery would "pause" adding of hidden wallet. This still passed because we didn't have a timeout on action. But recently we introduced a timeout for actions. So we need to properly handle this situation. And that means waiting for discovery between enabling coins and adding hidden wallet

Also improve Cardano staking validation. Recent changes made if unstable. And I was told there will be more changes so switched the validation from Aria snapshot to less broad asserts.

### CODE REVIEW

File: packages/suite-desktop-core/e2e/tests/trading/swap-coin-to-token.test.ts [hunk 1]
```diff
@@ -35,6 +35,8 @@ test.describe('Trading - Swap coin to token', { tag: ['@group=other', '@webOnly'
             await settingsPage.coins.enableNetwork('sol');
             await settingsPage.coins.enableNetwork('eth');
             await settingsPage.coins.disableNetwork('btc');
+            await settingsPage.coins.activateCoinsButton.click();
+            await dashboardPage.discoveryShouldFinish();
```
[the remark this answers is not in the document]
Comment (Vere-Grey) [reply packages/suite-desktop-core/e2e/tests/trading/swap-coin-to-token.test.ts:39]: That is correct, the bticoin is discovered by default.

**Review by HajekOndrej — APPROVED** [review]
